### PR TITLE
Bottom navigation items length docs

### LINKED ISSUE #72443: Requirement for 2 items in bottom navigation bar is not documented
https://github.com/flutter/flutter/blob/78910062997c3a836feee883712c241a5fd22983/packages/flutter/lib/src/material/bottom_navigation_bar.dart#L196 causes an assert when a bottom navigation bar is created.  However, https://api.flutter.dev/flutter/material/BottomNavigationBar-class.html doesn't mention this requirement.

The sentence "A material widget that's displayed at the bottom of an app for selecting among a small number of views, typically between three and five." should be expanded to indicated that at least two are needed for this Widget.

(Or, reduce the requirement for one, as might be useful in the case where only "Home" is desired")

### PR DESCRIPTION
**Pull request by Abhishek01039**
## Description

Fixes https://github.com/flutter/flutter/issues/72443

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR. There should be at least one issue listed here.*

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behavior with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by Hixie — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 1]
```diff
@@ -51,6 +51,9 @@ enum BottomNavigationBarType {
 /// [BottomNavigationBarType.fixed] when there are less than four items, and
 /// [BottomNavigationBarType.shifting] otherwise.
 ///
+/// The length of [items] must be at least two and each item's icon and title
```
Comment (yash1200) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart:54]: I guess title/label should be more convenient.

Comment (Abhishek01039) [reply packages/flutter/lib/src/material/bottom_navigation_bar.dart]: Done Thanks @yash1200

**Comment by yash1200** [comment]
Rebase your branch with master, this will fix these test fails

**Comment by Abhishek01039** [comment]
Sorry, but how can I do that?

**Pull request merged**